### ISSUE TITLE
Marketplace: Begin automating plugin adding

### ISSUE THREAD

**Issue by allilevine** [Automattic/wp-calypso#62268]
Let's make adding a plugin easier. What steps can we automate?

Acceptance criteria:
- [x] A list of what can be automated
- [x] A plan for automating them
- [x] Begin implementing the plan.

**Comment by lsl** [comment]
>  Implement the plan.

Can we make this part a stretch? Or at least rewrite as "begin implementation"? Thinking we might find some quick wins but a full implementation isn't likely.

**Comment by allilevine** [comment]
> Can we make this part a stretch? Or at least rewrite as "begin implementation"? Thinking we might find some quick wins but a full implementation isn't likely.

@lsl Definitely, thanks! I changed the issue title and that step.

**Comment by allilevine** [comment]
What done will look like: p2 what we can and can't automate. @cpapazoglou Let us know your ideas and thoughts :)

**Comment by cpapazoglou** [comment]
@allilevine, @lsl is this a duplicate of https://github.com/Automattic/wp-calypso/issues/61531? 

@allilevine did you mean to add a P2 link?

**Comment by allilevine** [comment]
> @allilevine, @lsl is this a duplicate of #61531?

Maybe, thanks! This might be more of a sub-task. 

> @allilevine did you mean to add a P2 link?

I meant we would write a P2 about what we can and can't automate. I updated my note.

**Comment by cpapazoglou** [comment]
Cool, my impression is that we can tackle this in two phases:
1) what we can automate now so that the process is faster and more robust (this is what we are currently doing semi-manually PCYsg-CJy-p2)
2) what we will automate in the future so that products can be provisioned (created, prices updated) from WC.com

I believe we should work closely with @Automattic/shilling team to identify how we can achieve (1).

FWIW, I don't believe we should work in automating the step that adds the plugin in the endpoint since we will soon be switching to WC.com for getting these results. The end goal is:
- Vendor adds a product in WC.com
- (manually) We approve it after Atomic sets it as managed plugin
- (automatically) Product is displayed in WordPress.com ready for purchase

**Comment by lsl** [comment]
Got some rough thoughts, will p2 + cc shilling tomorrow.

**Comment by lsl** [comment]
pdh6GB-zZ-p2

**Comment by lsl** [comment]
Waiting on more feedback before nailing anything down but I think we've got a much clearer picture of what needs doing:

1. Move code maps to a db mapping table.
2. Create marketplace specific product addition commands that use the mapping table instead of expecting hard coded mapping.
3. Run the new commands via async jobs triggered by wccom hooks.

- [x] A list of what can be automated - see pdh6GB-zZ-p2
- [x] A plan for automating them - see above/comments on p2
- [x] Begin implementing the plan - technically we did this by streamlining the docs (I noted this in the p2 post :wink:)
